### ISSUE TITLE
How to run pre-training?

### ISSUE THREAD

**Issue by BlessedTatonka** [AnswerDotAI/ModernBERT#155]
Thank you for the great work! I have a question regarding pre-training. Could you please clarify which YAML configuration file should be used to achieve a similar pre-training setup as ModernBert, but for a different language? I noticed that in the yamls folder, there doesn’t seem to be a specific file for this purpose. The only related script I found is generate_eval_config.py, which, if I understand correctly, generates a YAML configuration using ModernBert’s training params. Is my understanding correct, or am I missing something?

**Comment by yzimmermann** [comment]
Was wondering about this, too!

**Comment by chaofan520** [comment]
same question

**Comment by GithubX-F** [comment]
same question.
Are there any examples provided?

**Comment by NohTow** [comment]
Hello,
Sorry for the delayed response.
We plan to write proper guides to run the pretraining in the next days, we have been a bit short on time lately.
In the mean time, I have dropped the configs for the first step of pretraining (warmup+stable phases) [here](https://github.com/AnswerDotAI/ModernBERT/tree/pretraining_documentation/yamls/main) if you want to give it a shot before we clean up everything.

Note that the data path should point to data folder in the [MDS format](https://docs.mosaicml.com/projects/streaming/en/latest/preparing_datasets/dataset_format.html#:~:text=1.-,MDS,sample's%20entry%20for%20that%20column.), you have and example with the C4 dataset [here](https://github.com/mosaicml/examples/tree/main/examples/benchmarks/bert#prepare-your-data).
Also note that, to use streaming: True, you might need to decompress the data using [this script](https://github.com/AnswerDotAI/ModernBERT/blob/9e997569ea015b7710db413db01eb217cb2ab201/src/data/mds_conversion.py). Disabling streaming makes pretraining faster and solves an issue with uneven GPUs memory allocation (see #85).

Again, sorry for the delay and hopefully we'll have better documentation soon.

**Comment by BramVanroy** [comment]
@NohTow Hi there! Any update on a proper step-by-step guide for pretraining?

**Comment by NohTow** [comment]
Hello,

Until we update the readmes and merge the config, the above comment is the closest thing to a step-by-step guide.
I agree that this is not optimal for now, and I again apologies for the delay, but could you specify what information are you lacking w.r.t the comment so we can add it to the readme?
Thanks!

**Comment by ebrarkiziloglu** [comment]
Hi @NohTow, following your guide, I am encountering an issue in pre-training. Could you help with #199 ?
